crates/nu-cli/src/completion/matchers/unicode_case_insensitive.rs

```diff
@@ -0,0 +1,16 @@
+use unicase::UniCase;
+use crate::completion::matchers;
+pub struct Matcher;
+
+impl matchers::Matcher for Matcher {
```

Comment:
Personally, I think having multiple "case insensitive" options could be confusing to a user. Is there any reason we need both this one and naive? Could we just have one? Specifically, I'm thinking we just have "case insensitive" and it's this one, which I expect is a "superset" of the naive implementation?

Reply:
so the unicode matcher does unicode case folding, If you are happy with pulling in UniCase for this, then I can remove the naive one.

It may not be worth it, as most terminal usage is ascii driven? I'll let you guys choos.e

Reply:
But yeah, give me the word and I will make the remove the naive implementation

Reply:
> It may not be worth it, as most terminal usage is ascii driven? I'll let you guys choose

I agree with you here, that it's probably not worth it, at least not right now. Let's go with the naive implementation in this PR, and switch to something like `unicase` when someone puts in a feature request for it.